testutil/fuse/fuse.go

```diff
@@ -0,0 +1,238 @@
+	return h.fs.Access(name, mode, context)
+}
+
+func (h *HookFs) Link(oldName string, newName string, context *fuse.Context) fuse.Status {
+	return h.fs.Link(oldName, newName, context)
+}
+
+func (h *HookFs) Mkdir(name string, mode uint32, context *fuse.Context) fuse.Status {
+	return h.fs.Mkdir(name, mode, context)
+}
+
+func (h *HookFs) Mknod(name string, mode uint32, dev uint32, context *fuse.Context) fuse.Status {
+	return h.fs.Mknod(name, mode, dev, context)
+}
+
+func (h *HookFs) Rename(oldName string, newName string, context *fuse.Context) fuse.Status {
+	preHooked, err := h.hook.PreRename(oldName, newName)
+	if preHooked {
+		if err != nil {
+
+			return fuse.ToStatus(err)
+		}
+	}
+
+	status := h.fs.Rename(oldName, newName, context)
+
+	postHooked, err := h.hook.PostRename(oldName, newName)
+	if postHooked {
+		if err != nil {
+			return fuse.ToStatus(err)
+		}
+	}
+	return status
+}
+
+func (h *HookFs) Rmdir(name string, context *fuse.Context) fuse.Status {
+	return h.fs.Rmdir(name, context)
+}
+
+func (h *HookFs) Unlink(name string, context *fuse.Context) fuse.Status {
+	return h.fs.Unlink(name, context)
+}
+
+func (h *HookFs) GetXAttr(name string, attribute string, context *fuse.Context) ([]byte, fuse.Status) {
+	return h.fs.GetXAttr(name, attribute, context)
+}
+
+func (h *HookFs) ListXAttr(name string, context *fuse.Context) ([]string, fuse.Status) {
+	return h.fs.ListXAttr(name, context)
+}
+
+func (h *HookFs) RemoveXAttr(name string, attr string, context *fuse.Context) fuse.Status {
+	return h.fs.RemoveXAttr(name, attr, context)
+}
+
+func (h *HookFs) SetXAttr(name string, attr string, data []byte, flags int, context *fuse.Context) fuse.Status {
+	return h.fs.SetXAttr(name, attr, data, flags, context)
+}
+
+func (h *HookFs) OnMount(nodeFs *pathfs.PathNodeFs) {
+	h.fs.OnMount(nodeFs)
+}
+
+func (h *HookFs) OnUnmount() {
+	h.fs.OnUnmount()
+}
+
+func (h *HookFs) Open(name string, flags uint32, context *fuse.Context) (nodefs.File, fuse.Status) {
+	return h.fs.Open(name, flags, context)
+}
+
+func (h *HookFs) Create(name string, flags uint32, mode uint32, context *fuse.Context) (nodefs.File, fuse.Status) {
+	return h.fs.Create(name, flags, mode, context)
+}
+
+func (h *HookFs) OpenDir(name string, context *fuse.Context) ([]fuse.DirEntry, fuse.Status) {
+	return h.fs.OpenDir(name, context)
+}
+
+func (h *HookFs) Symlink(value string, linkName string, context *fuse.Context) fuse.Status {
+	return h.fs.Symlink(value, linkName, context)
+}
+
+func (h *HookFs) Readlink(name string, context *fuse.Context) (string, fuse.Status) {
+	return h.fs.Readlink(name, context)
+}
+
+func (h *HookFs) StatFs(name string) *fuse.StatfsOut {
+	return h.fs.StatFs(name)
+}
+
+func (h *HookFs) NewServe() (*fuse.Server, error) {
+	server, err := newHookServer(h)
+	if err != nil {
+		return nil, err
+	}
+
+	return server, nil
+}
+
+//tests will want to run this in a  goroutine.
```

Comment:
I think I mentioned it few times, but all comments should start with a capital letter and end with a full stop.

Also all exported objects should have a comment following the golang rules.

Reply:
Thanks for your patience,  I will bear it in mind. I used golang language for 3 months and I learned a lot from you. Thanks very much.

Reply:
Done, please review it again.
Thanks